src/main/java/core/basesyntax/strategy/handler/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        int fruitsBefore = FruitStorage.getStorage().get(transaction.getFruit());
+        int fruitsToRemove = transaction.getQuantity();
+        if (fruitsBefore < fruitsToRemove) {
+            throw new RuntimeException("Not enough fruits in storage");
```

Comment:
```suggestion
            throw new RuntimeException("Not enough fruits in storage: " + fruitsBefore);
```

Reply:
Maybe also try to get more clear variable name (it can be "balance", for example)